webapp/channels/src/components/analytics/doughnut_chart.tsx

```diff
@@ -16,30 +17,56 @@ type Props = {
 const DoughnutChart: React.FC<Props> = ({title, width, height, data}) => {
     const canvasRef = useRef<HTMLCanvasElement | null>(null);
     const chartRef = useRef<Chart<'doughnut'> | null>(null);
+    const prevData = useRef(data);
 
     useEffect(() => {
-        if (!data || !canvasRef.current) {
-            return;
-        }
+        const isEqual = data === prevData.current || deepEqual(prevData.current, data);
+
+        if (!isEqual) {
```

Comment:
You can make the function shallower by early returning. Instead of...

```
{
    ...
    if (!isEqual) {
        ...
    }
}
```
You can do
```
{
    ...
    if (isEqual) {
        return;
    }
    ...
}
```

Reply:
Done!